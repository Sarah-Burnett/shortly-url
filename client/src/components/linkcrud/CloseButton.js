import React from 'react';
import { Button } from '../Styles';
import { useProcessLink } from '../../hooks/useProcessLink';

const CloseButton = () => {
    const { setProcessNull } = useProcessLink();
    return (
        <Button type="button" onClick={setProcessNull}>Close</Button>
    )
};

export default CloseButton;

