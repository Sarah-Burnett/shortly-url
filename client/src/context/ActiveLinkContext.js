import React, { useState, createContext } from 'react';

export const ActiveLinkContext = createContext();

export const ActiveLinkProvider = (props) => {
    const [ activeLink, setActiveLink] = useState({
        activeLink: [],
        process: false
    });

    return (
        <ActiveLinkContext.Provider value={[activeLink, setActiveLink]}>
            {props.children}
        </ActiveLinkContext.Provider>
    )
};