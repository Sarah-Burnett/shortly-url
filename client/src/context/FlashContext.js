import React, { useState, createContext } from 'react';

export const FlashContext = createContext();

export const FlashProvider = (props) => {
    const [ flash, setFlash ] = useState({
        flashMsg: "",
        isFlashVisible: false,
        isAutoHideFlash: true,
    });
    return (
        <FlashContext.Provider value={[flash, setFlash]}>
            {props.children}
        </FlashContext.Provider>
    )
};