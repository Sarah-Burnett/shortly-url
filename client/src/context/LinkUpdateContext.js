import React, { useState, createContext } from 'react';

export const LinkUpdateContext = createContext();

export const LinkUpdateProvider = (props) => {
    const [ linkUpdate, setLinkUpdate] = useState();
    return (
        <LinkUpdateContext.Provider value={[linkUpdate, setLinkUpdate]}>
            {props.children}
        </LinkUpdateContext.Provider>
    )
};