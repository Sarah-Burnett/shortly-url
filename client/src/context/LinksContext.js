import React, { useState, createContext } from 'react';

export const LinksContext = createContext();

export const LinksProvider = (props) => {
    const [ links, setLinks] = useState([]);

    return (
        <LinksContext.Provider value={[links, setLinks]}>
            {props.children}
        </LinksContext.Provider>
    )
};