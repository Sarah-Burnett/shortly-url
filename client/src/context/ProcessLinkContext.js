import React, { useState, createContext } from 'react';

export const ProcessLinkContext = createContext();

export const ProcessLinkProvider = (props) => {
    const [ processLink, setProcessLink] = useState({
        link: [],
        process: false
    });
    return (
        <ProcessLinkContext.Provider value={[processLink, setProcessLink]}>
            {props.children}
        </ProcessLinkContext.Provider>
    )
};