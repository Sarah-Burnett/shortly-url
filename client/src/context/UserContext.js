import React, { useState, createContext } from 'react';

export const UserContext = createContext();

export const UserProvider = (props) => {
    const [ user, setUser] = useState({
        isAuth: false,
        firstName: undefined,
        id: undefined,
        token: undefined,
    });
    
    return (
        <UserContext.Provider value={[user, setUser]}>
            {props.children}
        </UserContext.Provider>
    )
};