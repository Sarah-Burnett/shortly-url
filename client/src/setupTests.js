import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent, screen, waitForElement } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import renderer from 'react-test-renderer';
import { UserProvider } from './context/UserContext';
import { LinksProvider } from './context/LinksContext';
import { ActiveLinkProvider } from './context/ActiveLinkContext';
import { ProcessLinkProvider } from './context/ProcessLinkContext';
import { FlashProvider } from './context/FlashContext';
import App from './App';

export const ComponentWrapper = (Component) => {
    return (
        <UserProvider>
            <LinksProvider>
                <ActiveLinkProvider>
                    <ProcessLinkProvider>
                        <FlashProvider>
                            <Component/>
                        </FlashProvider>
                    </ProcessLinkProvider>
                </ActiveLinkProvider>
            </LinksProvider>
        </UserProvider>
    )
}

export const rendersWithoutCrashing = (Component) => {
    const div = document.createElement('div');
    ReactDOM.render(<Component/>, div);
}

export const matchesSnapshot = (Component) => {
    const tree = renderer.create(<Component/>).toJSON();
    expect(tree).toMatchSnapshot();
}

export const sampleLinks = [
    {
        "date":"2020-08-14T17:29:53.041Z",
        "click":[
            {"date":1597426302926},
            {"date":1597426306071},
            {"date":1597426310231},
            {"date":1597426314763},
            {"date":1597428206995},
            {"date":1597428210093},
            {"date":1597428215131},
            {"date":1597428218755}
        ],
        "_id":"K1pjMCK-F",
        "full":"http://reactcommunity.org/react-transition-group/css-transition",
        "userId":"5ebe7d235e4a501ee8699058",
        "short":"octopus","__v":8
    },
    {
        "date":"2020-08-14T17:33:55.748Z",
        "click":[],
        "_id":"lsncCe5RV",
        "full":"https://ba-grant.netlify.app/",
        "userId":"5ebe7d235e4a501ee8699058",
        "short":"bagrant","__v":0
    }
];
