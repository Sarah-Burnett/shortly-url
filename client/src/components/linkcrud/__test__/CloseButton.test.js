import React from "react";
import CloseButton from "../CloseButton";
import { rendersWithoutCrashing, matchesSnapshot } from "../../../setupTests";
import { ProcessLinkProvider } from "../../../context/ProcessLinkContext";

const CloseButtonWrapper = () => {
	return (
		<ProcessLinkProvider>
			<CloseButton />
		</ProcessLinkProvider>
	);
};

it("renders close button without crashing", () =>
	rendersWithoutCrashing(CloseButtonWrapper));

it("close button matches snapshot", () => matchesSnapshot(CloseButtonWrapper));
