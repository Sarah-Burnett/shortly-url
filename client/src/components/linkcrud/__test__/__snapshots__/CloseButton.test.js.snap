// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`close button matches snapshot 1`] = `
<button
  className="sc-AxjAm iwCWZU"
  onClick={[Function]}
  type="button"
>
  Close
</button>
`;
